### PR TITLE
Windup 793 + Windup 815 Compatible report refactoring

### PR DESCRIPTION
**Pull request by mbriskar**
1.) Introduced default boolean value for the property
```
@FrameBooleanDefaultValue(false)
    @Property(WINDUP_GENERATED)
    void setWindupGenerated(boolean generated);
```
This is just for the boolean value to see, if it will be useful in the future and other values may be easily provided. This will set the default value for the given property

2.) FileModel.isWindupGenerated() -> tells if the file was generated by windup or if it originates from the application

3.) Changed compatible report to list only the files that do not have hint/classif., files whose decompiled .java file also doesn't have hint/classif. Does not list generated/decompiled files

### CODE REVIEW

**Comment by windupgithubbot1** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://wonka.mw.lab.eng.bos.redhat.com/jenkins//job/windup-pr-builder/1155/<h2>Failed Tests: <span class='status-failure'>2</span></h2><h3><a name='windup-pr-builder/org.jboss.windup.reporting:windup-reporting-tests' /><a href='http://wonka.mw.lab.eng.bos.redhat.com/jenkins/job/windup-pr-builder/1155/org.jboss.windup.reporting$windup-reporting-tests/testReport'>windup-pr-builder/org.jboss.windup.reporting:windup-reporting-tests</a>: <span class='status-failure'>1</span></h3><ul><li><a href='http://wonka.mw.lab.eng.bos.redhat.com/jenkins/job/windup-pr-builder/1155/org.jboss.windup.reporting$windup-reporting-tests/testReport/org.jboss.windup.reporting/InlineHintModelQueryTest/testFindingNonClassifiedFiles/'><strong>org.jboss.windup.reporting.InlineHintModelQueryTest.testFindingNonClassifiedFiles</strong></a></li></ul><h3><a name='windup-pr-builder/org.jboss.windup.tests:windup-tests' /><a href='http://wonka.mw.lab.eng.bos.redhat.com/jenkins/job/windup-pr-builder/1155/org.jboss.windup.tests$windup-tests/testReport'>windup-pr-builder/org.jboss.windup.tests:windup-tests</a>: <span class='status-failure'>1</span></h3><ul><li><a href='http://wonka.mw.lab.eng.bos.redhat.com/jenkins/job/windup-pr-builder/1155/org.jboss.windup.tests$windup-tests/testReport/org.jboss.windup.tests.application/WindupArchitectureMediumBinaryModeTest/testRunWindupMedium/'><strong>org.jboss.windup.tests.application.WindupArchitectureMediumBinaryModeTest.testRunWindupMedium</strong></a></li></ul>

**Comment by windupgithubbot1** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://wonka.mw.lab.eng.bos.redhat.com/jenkins//job/windup-core-PR-dependents-check/527/

**Comment by windupgithubbot1** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://wonka.mw.lab.eng.bos.redhat.com/jenkins//job/windup-pr-builder/1157/<h2>Failed Tests: <span class='status-failure'>1</span></h2><h3><a name='windup-pr-builder/org.jboss.windup.tests:windup-tests' /><a href='http://wonka.mw.lab.eng.bos.redhat.com/jenkins/job/windup-pr-builder/1157/org.jboss.windup.tests$windup-tests/testReport'>windup-pr-builder/org.jboss.windup.tests:windup-tests</a>: <span class='status-failure'>1</span></h3><ul><li><a href='http://wonka.mw.lab.eng.bos.redhat.com/jenkins/job/windup-pr-builder/1157/org.jboss.windup.tests$windup-tests/testReport/org.jboss.windup.tests.application/WindupArchitectureMediumBinaryModeTest/testRunWindupMedium/'><strong>org.jboss.windup.tests.application.WindupArchitectureMediumBinaryModeTest.testRunWindupMedium</strong></a></li></ul>

**Comment by windupgithubbot1** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://wonka.mw.lab.eng.bos.redhat.com/jenkins//job/windup-core-PR-dependents-check/529/

**Comment by mbriskar** [comment]
This one works on my machine, not sure why it throws exception

**Comment by mbriskar** [comment]
retest this please

**Comment by windupgithubbot1** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://wonka.mw.lab.eng.bos.redhat.com/jenkins//job/windup-pr-builder/1161/<h2>Failed Tests: <span class='status-failure'>1</span></h2><h3><a name='windup-pr-builder/org.jboss.windup.tests:windup-tests' /><a href='http://wonka.mw.lab.eng.bos.redhat.com/jenkins/job/windup-pr-builder/1161/org.jboss.windup.tests$windup-tests/testReport'>windup-pr-builder/org.jboss.windup.tests:windup-tests</a>: <span class='status-failure'>1</span></h3><ul><li><a href='http://wonka.mw.lab.eng.bos.redhat.com/jenkins/job/windup-pr-builder/1161/org.jboss.windup.tests$windup-tests/testReport/org.jboss.windup.tests.application/WindupArchitectureMediumBinaryModeTest/testRunWindupMedium/'><strong>org.jboss.windup.tests.application.WindupArchitectureMediumBinaryModeTest.testRunWindupMedium</strong></a></li></ul>

**Comment by windupgithubbot1** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://wonka.mw.lab.eng.bos.redhat.com/jenkins//job/windup-core-PR-dependents-check/533/

**Comment by windupgithubbot1** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://wonka.mw.lab.eng.bos.redhat.com/jenkins//job/windup-core-PR-dependents-check/534/

**Comment by windupgithubbot1** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://wonka.mw.lab.eng.bos.redhat.com/jenkins//job/windup-pr-builder/1162/<h2>Failed Tests: <span class='status-failure'>1</span></h2><h3><a name='windup-pr-builder/org.jboss.windup.tests:windup-tests' /><a href='http://wonka.mw.lab.eng.bos.redhat.com/jenkins/job/windup-pr-builder/1162/org.jboss.windup.tests$windup-tests/testReport'>windup-pr-builder/org.jboss.windup.tests:windup-tests</a>: <span class='status-failure'>1</span></h3><ul><li><a href='http://wonka.mw.lab.eng.bos.redhat.com/jenkins/job/windup-pr-builder/1162/org.jboss.windup.tests$windup-tests/testReport/org.jboss.windup.tests.application/WindupArchitectureMediumBinaryModeTest/testRunWindupMedium/'><strong>org.jboss.windup.tests.application.WindupArchitectureMediumBinaryModeTest.testRunWindupMedium</strong></a></li></ul>

**Comment by windupgithubbot1** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://wonka.mw.lab.eng.bos.redhat.com/jenkins//job/windup-pr-builder/1167/

**Comment by windupgithubbot1** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://wonka.mw.lab.eng.bos.redhat.com/jenkins//job/windup-core-PR-dependents-check/539/

File: graph/api/src/main/java/org/jboss/windup/graph/model/resource/FileModel.java [hunk 1]
```diff
@@ -235,6 +250,16 @@ public String getPrettyPath()
             return result;
         }
 
+        public Boolean isWindupGenerated()
```
Comment (mbriskar) [inline graph/api/src/main/java/org/jboss/windup/graph/model/resource/FileModel.java]: this is not needed anymore thanks to Default initilaization

**Comment by windupgithubbot1** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://wonka.mw.lab.eng.bos.redhat.com/jenkins//job/windup-pr-builder/1179/<h2>Failed Tests: <span class='status-failure'>1</span></h2><h3><a name='windup-pr-builder/org.jboss.windup.rexster:windup-rexster-tests' /><a href='http://wonka.mw.lab.eng.bos.redhat.com/jenkins/job/windup-pr-builder/1179/org.jboss.windup.rexster$windup-rexster-tests/testReport'>windup-pr-builder/org.jboss.windup.rexster:windup-rexster-tests</a>: <span class='status-failure'>1</span></h3><ul><li><a href='http://wonka.mw.lab.eng.bos.redhat.com/jenkins/job/windup-pr-builder/1179/org.jboss.windup.rexster$windup-rexster-tests/testReport/org.jboss.windup.graph.rexster.test.debug/RexsterDefaultDeploymentDebugTest/testRexsterProperStart/'><strong>org.jboss.windup.graph.rexster.test.debug.RexsterDefaultDeploymentDebugTest.testRexsterProperStart</strong></a></li></ul>

**Comment by windupgithubbot1** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://wonka.mw.lab.eng.bos.redhat.com/jenkins//job/windup-core-PR-dependents-check/551/

**Comment by jsight** [comment]
retest

**Comment by windupgithubbot1** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://wonka.mw.lab.eng.bos.redhat.com/jenkins//job/windup-pr-builder/1183/

**Comment by windupgithubbot1** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://wonka.mw.lab.eng.bos.redhat.com/jenkins//job/windup-core-PR-dependents-check/555/<h2>Build result: FAILURE</span></h2>[...truncated 48218 lines...][ERROR] -> [Help 1][ERROR] [ERROR] To see the full stack trace of the errors, re-run Maven with the -e switch.[ERROR] Re-run Maven using the -X switch to enable full debug logging.[ERROR] [ERROR] For more information about the errors and possible solutions, please read the following articles:[ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/MojoFailureException[ERROR] [ERROR] After correcting the problems, you can resume the build with the command[ERROR]   mvn <goals> -rf :windup-proprietary-javaee-servlet-rules-java[JENKINS] Archiving /var/lib/jenkins/workspace/windup-core-PR-dependents-check/pom.xml to org.jboss.windup.jenkins/windup-core-dependents-aggregate-pom/2.3.0-SNAPSHOT/windup-core-dependents-aggregate-pom-2.3.0-SNAPSHOT.pom[JENKINS] Archiving /var/lib/jenkins/workspace/windup-core-PR-dependents-check/windup-quickstarts/proprietary-javaee-servlet/rules-java/pom.xml to org.jboss.windup.quickstarts/windup-proprietary-javaee-servlet-rules-java/2.5.0-SNAPSHOT/windup-proprietary-javaee-servlet-rules-java-2.5.0-SNAPSHOT.pom[JENKINS] Archiving /var/lib/jenkins/workspace/windup-core-PR-dependents-check/windup-rulesets/pom.xml to org.jboss.windup.rules/windup-rulesets/2.5.0-SNAPSHOT/windup-rulesets-2.5.0-SNAPSHOT.pom[JENKINS] Archiving /var/lib/jenkins/workspace/windup-core-PR-dependents-check/windup-rulesets/target/windup-rulesets-2.5.0-SNAPSHOT.jar to org.jboss.windup.rules/windup-rulesets/2.5.0-SNAPSHOT/windup-rulesets-2.5.0-SNAPSHOT.jar[JENKINS] Archiving /var/lib/jenkins/workspace/windup-core-PR-dependents-check/windup-rulesets/target/windup-rulesets-2.5.0-SNAPSHOT-sources.jar to org.jboss.windup.rules/windup-rulesets/2.5.0-SNAPSHOT/windup-rulesets-2.5.0-SNAPSHOT-sources.jar[JENKINS] Archiving /var/lib/jenkins/workspace/windup-core-PR-dependents-check/windup-quickstarts/proprietary-javaee-servlet/pom.xml to org.jboss.windup.quickstarts/windup-proprietary-javaee-servlet/2.5.0-SNAPSHOT/windup-proprietary-javaee-servlet-2.5.0-SNAPSHOT.pom[JENKINS] Archiving /var/lib/jenkins/workspace/windup-core-PR-dependents-check/windup-quickstarts/pom.xml to org.jboss.windup.quickstarts/windup-quickstarts-root/2.5.0-SNAPSHOT/windup-quickstarts-root-2.5.0-SNAPSHOT.pom[JENKINS] Archiving /var/lib/jenkins/workspace/windup-core-PR-dependents-check/windup-quickstarts/ejb-beanutils-async/rules-java/pom.xml to org.jboss.windup.quickstarts/windup-ejb-beanutils-async-rules-java/2.5.0-SNAPSHOT/windup-ejb-beanutils-async-rules-java-2.5.0-SNAPSHOT.pom[JENKINS] Archiving /var/lib/jenkins/workspace/windup-core-PR-dependents-check/windup-quickstarts/ejb-beanutils-async/pom.xml to org.jboss.windup.quickstarts/windup-ejb-beanutils-async/2.5.0-SNAPSHOT/windup-ejb-beanutils-async-2.5.0-SNAPSHOT.pom[JENKINS] Archiving /var/lib/jenkins/workspace/windup-core-PR-dependents-check/windup-quickstarts/victi.ms/pom.xml to org.jboss.windup.quickstarts/windup-victims/2.5.0-SNAPSHOT/windup-victims-2.5.0-SNAPSHOT.pomchannel stoppedArchiving artifactsIRC notifier plugin: Sending notification to: #windupSetting status of 76a69bc03df9b891b72b036c466d30fb00817762 to FAILURE with url http://wonka.mw.lab.eng.bos.redhat.com/jenkins/job/windup-core-PR-dependents-check/555/ and message: 'Build finished. No test results found.'

**Pull request merged**